dap_prinz_green_jobs/pipeline/ojo_application/flows/ojo_industry_measures_update.py

```diff
@@ -0,0 +1,197 @@
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument("--production", action="store_true", default=False)
+    parser.add_argument("--job_desc_column", default="description", type=str)
+    parser.add_argument("--id_column", default="id", type=str)
+    parser.add_argument("--test_n", default=1000, type=int)
+
+    args = parser.parse_args()
+    production = args.production
+    id_column = args.id_column
+    test_n = args.test_n
+    job_desc_column = args.job_desc_column
+
+    if not production:
+        chunk_size = 100
+    else:
+        chunk_size = 5000
+
+    print("loading datasets...")
+
+    green_ind_existing_data = pl.read_csv(
+        green_ind_existing_data_dir,
+        schema_overrides={
+            "SIC": pl.String,
+            "INDUSTRY GHG PER UNIT EMISSIONS": pl.String,
+        },
+    )
+
+    # Remove any job adverts which have existing green measures (there shouldn't really be any)
+    all_ojo_descriptions = pl.read_parquet(new_ojo_descriptions_dir)
+    existing_ids = set(green_ind_existing_data["job_id"].to_list())
+    new_ojo_descriptions = all_ojo_descriptions.filter(
+        ~pl.col("id").is_in(existing_ids)
+    )
+
+    print(
+        f"There are {len(new_ojo_descriptions)} job adverts without existing green industry measures (of which there are {len(existing_ids)})"
+    )
+
+    # The format used in IndustryMeasures
+    ojo_jobs_data = (
+        new_ojo_descriptions[[id_column, job_desc_column]]
+        .rename(
+            {
+                id_column: config["job_adverts"]["job_id_key"],
+                job_desc_column: config["job_adverts"]["job_text_key"],
+            }
+        )
+        .to_dicts()
+    )
+
+    if not production:
+        ojo_jobs_data = ojo_jobs_data[:test_n]
+
+    date_stamp = str(date.today().date()).replace("-", "")
+    folder_name = f"outputs/data/ojo_application/extracted_green_measures/{date_stamp}/"
+
+    inds_output_folder = f"outputs/data/green_industries_lists/{date_stamp}"
+
+    im = IndustryMeasures(
+        use_gpu=False, chunk_size=100
+    )  # for use in a EC2 instance set these to: use_gpu = True, chunk_size = 1000
+    im.load()
+
+    job_desc_chunks = list(partition_all(chunk_size, ojo_jobs_data))
+
+    t0 = time.time()
+    for i, job_desc_chunk in tqdm(enumerate(job_desc_chunks)):
+        ind_green_measures_dict = im.get_measures(job_desc_chunk)
+        save_to_s3(
+            BUCKET_NAME,
+            ind_green_measures_dict,
+            os.path.join(
+                inds_output_folder,
+                f"ojo_newest_industry_green_measures_production_{production}_interim/{i}.json",
+            ),
+        )
+
+    # Read them back in and save altogether
+    ind_measures_locs = get_s3_data_paths(
+        BUCKET_NAME,
+        os.path.join(
+            inds_output_folder,
+            f"ojo_newest_industry_green_measures_production_{production}_interim",
+        ),
+        file_types=["*.json"],
+    )
```

Comment:
@crispy-wonton here we do the processing of batches of 5000 job adverts (which takes a while to run) and save the outputs in interim files. Then at the end we read them all in together and save again.